packages/block-library/src/navigation-menu-item/edit.js

```diff
@@ -49,41 +77,41 @@ function NavigationMenuItemEdit( {
 	isParentOfSelectedBlock,
 	setAttributes,
 	insertMenuItemBlock,
+	fetchSearchSuggestions,
 } ) {
-	const [ isLinkOpen, setIsLinkOpen ] = useState( false );
-	const [ isEditingLink, setIsEditingLink ] = useState( false );
-	const [ urlInput, setUrlInput ] = useState( null );
+	const { label, opensInNewTab, title, url } = attributes;
+	const link = title ? { title, url } : null;
+	const [ isLinkOpen, setIsLinkOpen ] = useState( ! label && isSelected );
 
-	const inputValue = urlInput !== null ? urlInput : url;
+	/**
+	 * It's a kind of hack to handle closing the LinkControl popover
+	 * clicking on the ToolbarButton link.
+	 */
+	useEffect( () => {
+		if ( ! isSelected ) {
+			setIsLinkOpen( false );
+		}
+	}, [ isSelected ] );
+
+	/**
+	 * `onKeyDown` LinkControl handler.
+	 * It takes over to stop the event propagation to make the
+	 * navigation work, avoiding undesired behaviors.
+	 * For instance, it will block to move between menu items
+	 * when the LinkOver is focused.
```

Comment:
I think it might be a typo here, should it be `LinkControl`?

```suggestion
	 * when the LinkControl is focused.
```